vlasovsolver/cpu_trans_map_amr.cpp

```diff
@@ -334,103 +334,111 @@ bool trans_map_1d_amr(const dccrg::Dccrg<spatial_cell::SpatialCell,dccrg::Cartes
    int memsetTimerId = phiprof::initializeTimer("trans-amr-MemSet");
    int propagateTimerId = phiprof::initializeTimer("trans-amr-propagatePencil");
 
-#pragma omp parallel
+   std::vector<uint> activeBins;
+   for (auto& [bin, cells] : DimensionPencils[dimension].binsPencils) {
+      activeBins.push_back(bin);
+   }
+
+   #pragma omp parallel
    {
+      phiprof::Timer mappingTimer {mappingTimerId}; // mapping (top-level)
+
       // Vector of pointers to cell block data, used for both reading and writing
       std::vector<Vec> blockDataBuffer(DimensionPencils[dimension].sumOfLengths*WID3/VECL);
       std::vector<Realf*> cellBlockData(DimensionPencils[dimension].sumOfLengths);
       std::vector<uint> pencilBlocksCount(DimensionPencils[dimension].N);
 
       // Loop over velocity space blocks (threaded).
-#pragma omp for schedule(dynamic,1)
+      // Get global id of the velocity block
+      // Load data for pencils.
+      #pragma omp for schedule(dynamic,1) collapse(2)
       for(uint blocki = 0; blocki < unionOfBlocks.size(); blocki++) {
```

Comment:
Actually, I would recommend storing `unionOfBlocks.size()` and `activeBins.size()` in constant variables before this parallel region (or, placing `nActiveBins` in the pencils struct), so that the loops don't need to re-evaluate the container sizes every iteration.

Reply:
Are you sure? I think `vector::size` is just checking a variable, but there might indeed be some optimization if amount of iterations is definitely known before starting the loop

Reply:
I can't quote anything exact right now, but I have a feeling that at least at some point of the GPU development this had a noticeable impact. Probably connected with GPU page faulting. So perhaps completely unnecessary here. Worth trying perhaps, still?

Reply:
From OpenMP documentation: "Values of the loop control expressions must be the same for all iterations of the loop." I guess it can't hurt to explicitly make it so, but I think OpenMP assumes it's constant anyway